shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/service/ProcessingDownloaderServiceImpl.java

```diff
@@ -0,0 +1,209 @@
+            manageProcessingsDownload(processingList, downloadResults, zipOutputStream, format, withManifest, filesByAcquisitionId, converterId);
+
+            String ids = String.join(",", Stream.concat(processingList.stream().map(DatasetProcessing::getInputDatasets), processingList.stream().map(DatasetProcessing::getOutputDatasets)).map(dataset -> ((Dataset) dataset).getId().toString()).collect(Collectors.toList()));
+            ShanoirEvent event = new ShanoirEvent(ShanoirEventType.DOWNLOAD_DATASET_EVENT, ids,
+                    KeycloakUtil.getTokenUserId(), ids + "." + format, ShanoirEvent.IN_PROGRESS);
+            event.setStatus(ShanoirEvent.SUCCESS);
+            eventService.publishEvent(event);
+        } catch (Exception e) {
+            response.setContentType(null);
+            LOG.error("Unexpected error while downloading dataset files.", e);
+            throw new RestServiceException(
+                    new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(),
+                            "Unexpected error while downloading dataset files"));
+        }
+    }
+
+    private void manageProcessingsDownload(List<DatasetProcessing> processingList, Map<Long, DatasetDownloadError> downloadResults, ZipOutputStream zipOutputStream, String format, boolean withManifest, Map<Long, List<String>> filesByAcquisitionId, Long converterId) throws RestServiceException, IOException {
+        for (DatasetProcessing processing : processingList) {
+            String processingFilePath = getExecFilepath(processing.getId(), getExaminationDatas(processing.getInputDatasets()));
+            String subjectName = getProcessingSubject(processing);
+            for (Dataset dataset : Stream.concat(processing.getInputDatasets().stream(), processing.getOutputDatasets().stream()).toList()) {
+                manageDatasetDownload(dataset, downloadResults, zipOutputStream, subjectName, processingFilePath, format, withManifest, filesByAcquisitionId, converterId);
+
+            }
+        }
+        if(!filesByAcquisitionId.isEmpty()){
+            DatasetFileUtils.writeManifestForExport(zipOutputStream, filesByAcquisitionId);
+        }
+
+        // Write errors to the file
+        if (!downloadResults.isEmpty()) {
+            ZipEntry zipEntry = new ZipEntry(JSON_RESULT_FILENAME);
+            zipEntry.setTime(System.currentTimeMillis());
+            zipOutputStream.putNextEntry(zipEntry);
+            zipOutputStream.write(objectMapper.writeValueAsString(downloadResults).getBytes());
+            zipOutputStream.closeEntry();
+        }
+    }
+
+    private void manageDatasetDownload(Dataset dataset, Map<Long, DatasetDownloadError> downloadResults, ZipOutputStream zipOutputStream, String subjectName, String processingFilePath, String format, boolean withManifest, Map<Long, List<String>> filesByAcquisitionId, Long converterId) throws IOException, RestServiceException {
```

Comment:
is this method different to the existing already?

Reply:
Yes, the current code division in DatasetDownloaderServiceImpl does not allow to use this specific part of code. I would not change the existing code, that's why I did that way.

I did a specific commit for a code division update enabling to use that part of code in the children class, as I'm supposing you wanted it to be like that. Feel free to remove it or not at PR merging, it won't break the functionality